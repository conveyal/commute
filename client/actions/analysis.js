import moment from 'moment'
import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addAnalysis = createAction('add analysis')
export const createAnalysis = (newAnalysis, organizationId) => {
  newAnalysis.id = uuid.v4()
  newAnalysis.lastRunDateTime = moment().unix()
  const commuters = newAnalysis.commuters
  delete newAnalysis.commuters
  return [
    addAnalysis({ analysis: newAnalysis, organizationId }),
    push(`/organizations/${organizationId}/analysis/${newAnalysis.id}`),
    makeMockTrips(newAnalysis.id, organizationId, commuters)
  ] // TODO save to server
}

const deleteLocally = createAction('delete analysis')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/analysis/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteAnalysis = (id, organizationId) => [
  deleteLocally({ analysisId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]

const makeMockTrips = (analysisId, organizationId, commuters) => {
  const trips = []
  const modes = ['bike', 'car', 'transit', 'walk']
  const randRange = () => Math.random() * 0.4 + 0.8
  commuters.forEach((commuter) => {
    const curTrip = {
      commuterId: commuter.id
    }
    const tripMetricBase = Math.random()
    let bestValue = 9999999
    let bestMode
    modes.forEach((mode) => {
      // generate random trips
      const modePctMultiplier = randRange()
      const cost = tripMetricBase * 34.7 * modePctMultiplier * randRange()
      const distance = tripMetricBase * 69 * modePctMultiplier * randRange()
      const time = tripMetricBase * 7500 * modePctMultiplier * randRange()
      curTrip[mode] = {cost, distance, time}
      const tripValue = cost + time * 0.01
      if (tripValue < bestValue) bestMode = mode
    })
    curTrip.mostLikely = Object.assign({ mode: bestMode }, curTrip[bestMode])
    trips.push(curTrip)
  })
  return sendMockTrips({analysisId, organizationId, trips})
}
const sendMockTrips = createAction('receive mock calculated trips')
