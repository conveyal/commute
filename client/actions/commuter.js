import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addCommuter = createAction('add commuter')
export const createCommuter = ({commuter, groupId, organizationId}) => {
  commuter.id = uuid.v4()
  return [
    addCommuter({commuter, groupId, organizationId}),
    push(`/organizations/${organizationId}/groups/${groupId}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete commuter')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/commuter/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteCommuter = ({commuterId, groupId, organizationId}) => [
  deleteLocally({commuterId, groupId, organizationId}),
  push(`/organizations/${organizationId}/groups/${groupId}`)
]

const updateLocally = createAction('update commuter')
/* const updateOnServer = (id) =>
  serverAction({
    url: `/api/commuter/${id}`,
    options: {
      method: 'update'
    }
  }) */ // TODO update on server
export const updateCommuter = ({commuter, groupId, organizationId}) => [
  updateLocally({commuter, groupId, organizationId}),
  push(`/organizations/${organizationId}/groups/${groupId}`)
]
