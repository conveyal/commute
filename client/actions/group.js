import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addGroup = createAction('add group')
const appendCommuters = createAction('append commuters')
export const appendToGroup = (newCommuters, groupId, organizationId) => {
  return [
    appendCommuters({ commuters: newCommuters, groupId, organizationId }),
    push(`/organizations/${organizationId}/groups/${groupId}`)
  ] // TODO save to server
}
export const createGroup = (newGroup, organizationId) => {
  newGroup.id = uuid.v4()
  newGroup.allAddressesGeocoded = false
  return [
    addGroup({ group: newGroup, organizationId }),
    push(`/organizations/${organizationId}/groups/${newGroup.id}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete group')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/group/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteGroup = (id, organizationId) => [
  deleteLocally({ groupId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]
