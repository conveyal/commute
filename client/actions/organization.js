import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addLocally = createAction('add organization')
export const createOrganization = (newOrganization) => {
  newOrganization.id = uuid.v4()
  newOrganization.analyses = []
  newOrganization.analysesById = {}
  newOrganization.sites = []
  newOrganization.sitesById = {}
  newOrganization.groups = []
  newOrganization.groupsById = {}
  return [
    addLocally(newOrganization),
    push(`/organizations/${newOrganization.id}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete organization')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/organization/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteOrganization = (id) => [deleteLocally(id), push('/')]

export const updateOrganization = (organization) => [
  setLocally(organization),
  push(`/organizations/${organization.id}`)
]
/* const updateOnServer = (organization) =>
  serverAction({
    url: `/api/organization/${organization.id}`,
    options: {
      body: JSON.stringify(organization),
      method: 'put'
    }
  }) */  // TODO: update on server

export const setLocally = createAction('set organization')
