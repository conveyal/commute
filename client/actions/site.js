import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addSite = createAction('add site')
export const createSite = (newSite, organizationId) => {
  newSite.id = uuid.v4()
  return [
    addSite({ organizationId, site: newSite }),
    push(`/organizations/${organizationId}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete site')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/site/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteSite = (id, organizationId) => [
  deleteLocally({ siteId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]

const updateLocally = createAction('update site')
/* const updateOnServer = (id) =>
  serverAction({
    url: `/api/site/${id}`,
    options: {
      method: 'update'
    }
  }) */ // TODO update on server
export const updateSite = (site, organizationId) => [
  updateLocally({ organizationId, site }),
  push(`/organizations/${organizationId}`)
]
