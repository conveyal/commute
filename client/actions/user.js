import {createAction} from 'redux-actions'

// login / logout
export const login = createAction('log in')
export const logout = createAction('log out')
