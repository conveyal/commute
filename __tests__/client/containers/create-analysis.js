/* global describe, expect, it */

import {mount} from 'enzyme'
import {mountToJson} from 'enzyme-to-json'
import React from 'react'
import {Provider} from 'react-redux'

import {expectCreateAnalysis} from '../../test-utils/actions'
import {makeMockStore, mockStores} from '../../test-utils/mock-data.js'

import CreateAnalysis from '../../../client/containers/create-analysis'

describe('Container > CreateAnalysis', () => {
  it('Create Analysis View loads', () => {
    const mockStore = makeMockStore(mockStores.complexOrganization)

    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <CreateAnalysis
          params={{organizationId: '2'}}
          />
      </Provider>
    )
    expect(mountToJson(tree.find(CreateAnalysis))).toMatchSnapshot()
  })

  it('Create analysis', () => {
    // Given a logged-in user is viewing the Create Analysis View
    mockStores.complexOrganization.analyses = []
    mockStores.complexOrganization.analysesById = {}
    const mockStore = makeMockStore(mockStores.complexOrganization)

    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <CreateAnalysis
          params={{organizationId: '2'}}
          />
      </Provider>
    )

    // When the user fills out all of the required fields (Site, commuter group)
    // site
    tree.find('Select').first().props().onChange({ value: '1' })
    // group
    tree.find('Select').last().props().onChange({ value: '1' })

    // And the user submits the form
    tree.find('button').last().simulate('click')

    expectCreateAnalysis(mockStore.getActions())
  })
})
