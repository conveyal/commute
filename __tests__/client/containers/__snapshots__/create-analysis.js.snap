exports[`Container > CreateAnalysis Create Analysis View loads 1`] = `
<Connect(CreateAnalysis)
  params={
    Object {
      "organizationId": "2",
    }
  }>
  <CreateAnalysis
    create={[Function]}
    organization={
      Object {
        "analyses": Array [
          Object {
            "groupId": "1",
            "id": "1",
            "lastRunDateTime": 1477697490,
            "name": "An Analysis",
            "siteId": "1",
            "summary": Object {
              "avgDistance": 12.34,
              "avgTravelTime": 1234,
              "savingsPerTrip": 1.23,
              "savingsPerTripYear": 319.8,
              "savingsTotalPerDay": 1.23,
              "savingsTotalPerYear": 319.8,
            },
            "tripVals": Object {
              "bike": Object {
                "cost": Array [
                  0.12,
                ],
                "distance": Array [
                  23,
                ],
                "time": Array [
                  2345,
                ],
              },
              "car": Object {
                "cost": Array [
                  6.78,
                ],
                "distance": Array [
                  18,
                ],
                "time": Array [
                  1234,
                ],
              },
              "transit": Object {
                "cost": Array [
                  3.45,
                ],
                "distance": Array [
                  30,
                ],
                "time": Array [
                  3456,
                ],
              },
              "walk": Object {
                "cost": Array [
                  0,
                ],
                "distance": Array [
                  19,
                ],
                "time": Array [
                  6789,
                ],
              },
            },
            "trips": Array [
              Object {
                "bike": Object {
                  "cost": 0.12,
                  "distance": 23,
                  "polygon": "encoded",
                  "time": 2345,
                },
                "car": Object {
                  "cost": 6.78,
                  "distance": 18,
                  "polygon": "encoded",
                  "time": 1234,
                },
                "commuterId": "1",
                "mostLikely": Object {
                  "cost": 3.45,
                  "distance": 30,
                  "mode": "transit",
                  "polygon": "encoded",
                  "time": 3456,
                },
                "transit": Object {
                  "cost": 3.45,
                  "distance": 30,
                  "polygon": "encoded",
                  "time": 3456,
                },
                "walk": Object {
                  "cost": 0,
                  "distance": 19,
                  "polygon": "encoded",
                  "time": 6789,
                },
              },
            ],
          },
        ],
        "analysesById": Object {
          "1": Object {
            "groupId": "1",
            "id": "1",
            "lastRunDateTime": 1477697490,
            "name": "An Analysis",
            "siteId": "1",
            "summary": Object {
              "avgDistance": 12.34,
              "avgTravelTime": 1234,
              "savingsPerTrip": 1.23,
              "savingsPerTripYear": 319.8,
              "savingsTotalPerDay": 1.23,
              "savingsTotalPerYear": 319.8,
            },
            "tripVals": Object {
              "bike": Object {
                "cost": Array [
                  0.12,
                ],
                "distance": Array [
                  23,
                ],
                "time": Array [
                  2345,
                ],
              },
              "car": Object {
                "cost": Array [
                  6.78,
                ],
                "distance": Array [
                  18,
                ],
                "time": Array [
                  1234,
                ],
              },
              "transit": Object {
                "cost": Array [
                  3.45,
                ],
                "distance": Array [
                  30,
                ],
                "time": Array [
                  3456,
                ],
              },
              "walk": Object {
                "cost": Array [
                  0,
                ],
                "distance": Array [
                  19,
                ],
                "time": Array [
                  6789,
                ],
              },
            },
            "trips": Array [
              Object {
                "bike": Object {
                  "cost": 0.12,
                  "distance": 23,
                  "polygon": "encoded",
                  "time": 2345,
                },
                "car": Object {
                  "cost": 6.78,
                  "distance": 18,
                  "polygon": "encoded",
                  "time": 1234,
                },
                "commuterId": "1",
                "mostLikely": Object {
                  "cost": 3.45,
                  "distance": 30,
                  "mode": "transit",
                  "polygon": "encoded",
                  "time": 3456,
                },
                "transit": Object {
                  "cost": 3.45,
                  "distance": 30,
                  "polygon": "encoded",
                  "time": 3456,
                },
                "walk": Object {
                  "cost": 0,
                  "distance": 19,
                  "polygon": "encoded",
                  "time": 6789,
                },
              },
            ],
          },
        },
        "contact": "someone",
        "email": "abc@def.ghi",
        "groups": Array [
          Object {
            "allAddressesGeocoded": true,
            "commuters": Array [
              Object {
                "address": "4321 XYZ Boulevard",
                "email": "luke@warm.cold",
                "id": "1",
                "lat": 38.916089,
                "lng": -76.970221,
                "name": "Luke Warm",
              },
            ],
            "commutersById": Object {
              "1": Object {
                "address": "4321 XYZ Boulevard",
                "email": "luke@warm.cold",
                "id": "1",
                "lat": 38.916089,
                "lng": -76.970221,
                "name": "Luke Warm",
              },
            },
            "id": "1",
            "name": "Mock Group",
          },
        ],
        "groupsById": Object {
          "1": Object {
            "allAddressesGeocoded": true,
            "commuters": Array [
              Object {
                "address": "4321 XYZ Boulevard",
                "email": "luke@warm.cold",
                "id": "1",
                "lat": 38.916089,
                "lng": -76.970221,
                "name": "Luke Warm",
              },
            ],
            "commutersById": Object {
              "1": Object {
                "address": "4321 XYZ Boulevard",
                "email": "luke@warm.cold",
                "id": "1",
                "lat": 38.916089,
                "lng": -76.970221,
                "name": "Luke Warm",
              },
            },
            "id": "1",
            "name": "Mock Group",
          },
        },
        "id": "2",
        "logo_url": "https://placekitten.com/80/80",
        "main_url": "https://placekitten.com/",
        "name": "Mock Organization",
        "sites": Array [
          Object {
            "address": "123 ABC St",
            "id": "1",
            "lat": 38.8886,
            "lng": -77.043,
            "name": "Acme Corp",
            "radius": 1,
          },
        ],
        "sitesById": Object {
          "1": Object {
            "address": "123 ABC St",
            "id": "1",
            "lat": 38.8886,
            "lng": -77.043,
            "name": "Acme Corp",
            "radius": 1,
          },
        },
      }
    }
    params={
      Object {
        "organizationId": "2",
      }
    }>
    <Grid
      bsClass="container"
      componentClass="div"
      fluid={false}>
      <div
        className="container">
        <Row
          bsClass="row"
          componentClass="div">
          <div
            className="row">
            <Col
              bsClass="col"
              componentClass="div"
              xs={12}>
              <div
                className="col-xs-12">
                <h3>
                  <span>
                    Create Analysis
                  </span>
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    className="pull-right"
                    disabled={false}>
                    <button
                      className="pull-right btn btn-default"
                      disabled={false}
                      type="button">
                      <Link
                        onlyActiveOnIndex={false}
                        style={Object {}}
                        to="/organizations/2">
                        <a
                          onClick={[Function]}
                          style={Object {}}>
                          <Icon
                            type="arrow-left">
                            <i
                              className="fa fa-arrow-left fa-fw " />
                          </Icon>
                        </a>
                      </Link>
                    </button>
                  </Button>
                </h3>
                <form>
                  <FormGroup
                    bsClass="form-group"
                    controlId="site-control">
                    <div
                      className="form-group">
                      <ControlLabel
                        bsClass="control-label"
                        srOnly={false}>
                        <label
                          className="control-label"
                          htmlFor="site-control">
                          Site
                        </label>
                      </ControlLabel>
                      <Select
                        addLabelText="Add \"{label}\"?"
                        arrowRenderer={[Function]}
                        autosize={true}
                        backspaceRemoves={true}
                        backspaceToRemoveMessage="Press backspace to remove {label}"
                        clearAllText="Clear all"
                        clearValueText="Clear value"
                        clearable={true}
                        delimiter=","
                        disabled={false}
                        escapeClearsValue={true}
                        filterOptions={[Function]}
                        ignoreAccents={true}
                        ignoreCase={true}
                        inputProps={Object {}}
                        isLoading={false}
                        joinValues={false}
                        labelKey="label"
                        matchPos="any"
                        matchProp="any"
                        menuBuffer={0}
                        menuRenderer={[Function]}
                        multi={false}
                        noResultsText="No results found"
                        onBlurResetsInput={true}
                        onChange={[Function]}
                        onCloseResetsInput={true}
                        openAfterFocus={false}
                        optionComponent={[Function]}
                        options={
                          Array [
                            Object {
                              "label": "Acme Corp",
                              "value": "1",
                            },
                          ]
                        }
                        pageSize={5}
                        placeholder="Select a Site..."
                        required={false}
                        scrollMenuIntoView={true}
                        searchable={true}
                        simpleValue={false}
                        tabSelectsValue={true}
                        value={undefined}
                        valueComponent={[Function]}
                        valueKey="value">
                        <div
                          className="Select Select--single is-searchable"
                          style={undefined}>
                          <div
                            className="Select-control"
                            onKeyDown={[Function]}
                            onMouseDown={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            style={undefined}>
                            <span
                              className="Select-multi-value-wrapper"
                              id="react-select-2--value">
                              <div
                                className="Select-placeholder">
                                Select a Site...
                              </div>
                              <AutosizeInput
                                aria-activedescendant="react-select-2--value"
                                aria-expanded="false"
                                aria-haspopup="false"
                                aria-label={undefined}
                                aria-labelledby={undefined}
                                aria-owns=""
                                className="Select-input"
                                minWidth="5px"
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                required={false}
                                role="combobox"
                                tabIndex={undefined}
                                value="">
                                <div
                                  className="Select-input"
                                  style={
                                    Object {
                                      "display": "inline-block",
                                    }
                                  }>
                                  <input
                                    aria-activedescendant="react-select-2--value"
                                    aria-expanded="false"
                                    aria-haspopup="false"
                                    aria-label={undefined}
                                    aria-labelledby={undefined}
                                    aria-owns=""
                                    className={undefined}
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    onFocus={[Function]}
                                    required={false}
                                    role="combobox"
                                    style={
                                      Object {
                                        "boxSizing": "content-box",
                                        "width": "5pxpx",
                                      }
                                    }
                                    tabIndex={undefined}
                                    value="" />
                                  <div
                                    style={
                                      Object {
                                        "height": 0,
                                        "left": 0,
                                        "overflow": "scroll",
                                        "position": "absolute",
                                        "top": 0,
                                        "visibility": "hidden",
                                        "whiteSpace": "pre",
                                      }
                                    } />
                                </div>
                              </AutosizeInput>
                            </span>
                            <span
                              className="Select-arrow-zone"
                              onMouseDown={[Function]}>
                              <span
                                className="Select-arrow"
                                onMouseDown={[Function]} />
                            </span>
                          </div>
                        </div>
                      </Select>
                    </div>
                  </FormGroup>
                  <FormGroup
                    bsClass="form-group"
                    controlId="group-control">
                    <div
                      className="form-group">
                      <ControlLabel
                        bsClass="control-label"
                        srOnly={false}>
                        <label
                          className="control-label"
                          htmlFor="group-control">
                          Group
                        </label>
                      </ControlLabel>
                      <Select
                        addLabelText="Add \"{label}\"?"
                        arrowRenderer={[Function]}
                        autosize={true}
                        backspaceRemoves={true}
                        backspaceToRemoveMessage="Press backspace to remove {label}"
                        clearAllText="Clear all"
                        clearValueText="Clear value"
                        clearable={true}
                        delimiter=","
                        disabled={false}
                        escapeClearsValue={true}
                        filterOptions={[Function]}
                        ignoreAccents={true}
                        ignoreCase={true}
                        inputProps={Object {}}
                        isLoading={false}
                        joinValues={false}
                        labelKey="label"
                        matchPos="any"
                        matchProp="any"
                        menuBuffer={0}
                        menuRenderer={[Function]}
                        multi={false}
                        noResultsText="No results found"
                        onBlurResetsInput={true}
                        onChange={[Function]}
                        onCloseResetsInput={true}
                        openAfterFocus={false}
                        optionComponent={[Function]}
                        options={
                          Array [
                            Object {
                              "label": "Mock Group",
                              "value": "1",
                            },
                          ]
                        }
                        pageSize={5}
                        placeholder="Select a Commuter Group..."
                        required={false}
                        scrollMenuIntoView={true}
                        searchable={true}
                        simpleValue={false}
                        tabSelectsValue={true}
                        value={undefined}
                        valueComponent={[Function]}
                        valueKey="value">
                        <div
                          className="Select Select--single is-searchable"
                          style={undefined}>
                          <div
                            className="Select-control"
                            onKeyDown={[Function]}
                            onMouseDown={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            style={undefined}>
                            <span
                              className="Select-multi-value-wrapper"
                              id="react-select-3--value">
                              <div
                                className="Select-placeholder">
                                Select a Commuter Group...
                              </div>
                              <AutosizeInput
                                aria-activedescendant="react-select-3--value"
                                aria-expanded="false"
                                aria-haspopup="false"
                                aria-label={undefined}
                                aria-labelledby={undefined}
                                aria-owns=""
                                className="Select-input"
                                minWidth="5px"
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                required={false}
                                role="combobox"
                                tabIndex={undefined}
                                value="">
                                <div
                                  className="Select-input"
                                  style={
                                    Object {
                                      "display": "inline-block",
                                    }
                                  }>
                                  <input
                                    aria-activedescendant="react-select-3--value"
                                    aria-expanded="false"
                                    aria-haspopup="false"
                                    aria-label={undefined}
                                    aria-labelledby={undefined}
                                    aria-owns=""
                                    className={undefined}
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    onFocus={[Function]}
                                    required={false}
                                    role="combobox"
                                    style={
                                      Object {
                                        "boxSizing": "content-box",
                                        "width": "5pxpx",
                                      }
                                    }
                                    tabIndex={undefined}
                                    value="" />
                                  <div
                                    style={
                                      Object {
                                        "height": 0,
                                        "left": 0,
                                        "overflow": "scroll",
                                        "position": "absolute",
                                        "top": 0,
                                        "visibility": "hidden",
                                        "whiteSpace": "pre",
                                      }
                                    } />
                                </div>
                              </AutosizeInput>
                            </span>
                            <span
                              className="Select-arrow-zone"
                              onMouseDown={[Function]}>
                              <span
                                className="Select-arrow"
                                onMouseDown={[Function]} />
                            </span>
                          </div>
                        </div>
                      </Select>
                    </div>
                  </FormGroup>
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="success"
                    disabled={false}
                    onClick={[Function]}>
                    <button
                      className="btn btn-success"
                      disabled={false}
                      onClick={[Function]}
                      type="button">
                      Create
                    </button>
                  </Button>
                </form>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Grid>
  </CreateAnalysis>
</Connect(CreateAnalysis)>
`;

exports[`Container > CreateAnalysis Create analysis 1`] = `
Object {
  "payload": Object {
    "analysis": Object {
      "groupId": "1",
      "siteId": "1",
    },
    "organizationId": "2",
  },
  "type": "add analysis",
}
`;

exports[`Container > CreateAnalysis Create analysis 2`] = `
Object {
  "payload": Object {
    "args": Array [
      "/organizations/2/analysis/new-entity-id",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;
